src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -1667,7 +1667,7 @@ public boolean isCastable(Type t, Type s, Warner warn) {
     }
     // where
         private boolean areDisjoint(ClassSymbol ts, ClassSymbol ss) {
-            if (isSubtype(ts.type, ss.type)) {
+            if (isSubtype(erasure(ts.type), erasure(ss.type))) {
```

Comment:
There is another `isSubtype` below - could you please check if it needs the same erasure handling?

Reply:
I agree, done